src/sync/branches.ts

```diff
@@ -3,18 +3,23 @@ import { getBranches as getBranchesQuery } from "./queries";
 import { GitHubAPI } from "probot";
 import { Repository } from "../models/subscription";
 import GitHubClient from "../github/client/github-client";
+import { booleanFlag, BooleanFlags } from "../config/feature-flags";
 
 // TODO: better typings
-export default async (github: GitHubAPI, _newGithub: GitHubClient, _jiraHost: string, repository:Repository, cursor?:string | number, perPage?:number) => {
+export default async (github: GitHubAPI, newGithub: GitHubClient, jiraHost: string, repository:Repository, cursor?:string | number, perPage?:number) => {
 	// TODO: fix typings for graphql
-	const { edges } = ((await github.graphql(getBranchesQuery, {
+	
+	const useNewGHClient = await booleanFlag(BooleanFlags.USE_NEW_GITHUB_CLIENT_FOR_BRANCHES, false, jiraHost);
+	const client = useNewGHClient ? newGithub : github;
+	const results = ((await client.graphql(getBranchesQuery, {
```

Comment:
Do we have a unit test for this? Would be good to add tests for the flag

Reply:
I would like to see the `getBranchesQuery` encapsulated in the `GitHubClient` (see my other comment) and instead call a function `getBranches()` here. This would also make it possible to add a unit test for the `getBranches()` function in the existing test for the GitHub client.

The `GitHubClient` should abstract away the fact that it's using GraphQL or REST under the hood.